include/ComputeTable.hpp

```diff
@@ -0,0 +1,84 @@
+/*
+ * This file is part of the JKQ DD Package which is released under the MIT license.
+ * See file README.md or go to http://iic.jku.at/eda/research/quantum_dd/ for more information.
+ */
+
+#ifndef DDpackage_COMPUTETABLE_HPP
+#define DDpackage_COMPUTETABLE_HPP
+
+#include "Definitions.hpp"
+
+#include <array>
+#include <functional>
+#include <iostream>
+
+namespace dd {
+
+    template<class A, class B, class R, std::size_t NBUCKET = 16384>
+    class ComputeTable {
+    public:
+        ComputeTable() = default;
+
+        struct Entry {
+            A a;
+            B b;
+            R r;
+        };
+
+        static constexpr std::size_t MASK = NBUCKET - 1; // must be CTSLOTS-1
+
+        static std::size_t hash(const A& a, const B& b) {
+            const auto h1   = std::hash<A>{}(a);
+            const auto h2   = std::hash<B>{}(b);
+            const auto hash = h1 ^ (h2 << 1);
+            return hash & MASK;
+        }
+
+        // access functions
+        [[nodiscard]] const auto& getTable() const { return table; }
+
+        void insert(const A& a, const B& b, const R& r) {
+            const auto key = hash(a, b);
+            table[key]     = {.a = a, .b = b, .r = r};
```

Comment:
Designated initializers are a C++20 feature :(

Reply:
Totally overlooked that. They are not important in these cases anyway 👍🏼